test/logger-errors.js

```diff
@@ -0,0 +1,385 @@
+  }))
+
+  logger.on('warn', t.fail)
+  logger.on('error', t.fail)
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger.url)
+    .post('/', (body) => {
+      const payload = body.ls
+      t.equal(payload.length, 2, 'Number of lines is correct')
+      const [{timestamp: stamp1}, {timestamp: stamp2}] = payload
+      t.ok(stamp1 > startTime, 'bad time format was replaced with a good one')
+      t.ok(stamp2 > startTime, 'out of range time was replaced with a good one')
+      return true
+    })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+    .persist()
+
+  logger.on('send', (obj) => {
+    t.deepEqual(obj, {
+      httpStatus: 200
+    , firstLine: 'my log line'
+    , lastLine: 'my log line'
+    , totalLinesSent: 2
+    , totalLinesReady: 0
+    , bufferCount: 0
+    }, 'Got send event')
+  })
+
+  logger.info('my log line', {
+    timestamp: {}
+  })
+  logger.info('my log line', {
+    timestamp: startTime + constants.MS_IN_A_DAY
+  })
+})
+
+test('.log() throws if options is a string and not a valid log entry', (t) => {
+  t.plan(2)
+  const logger = new Logger(apiKey, createOptions())
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger.url)
+    .post('/', () => { return true })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.on('error', (err) => {
+    t.type(err, TypeError, 'Expected to be a TypeError')
+    t.match(err, {
+      name: 'TypeError'
+    , message: 'If \'options\' is a string, then it must be a valid log level'
+    , meta: {
+        got: 'NOPE'
+      , expected: constants.LOG_LEVELS
+      }
+    }, 'Expected Error is correct')
+  })
+  logger.log('log line', 'NOPE')
+})
+
+test('.log() rejects invalid `opts` data type', (t) => {
+  t.plan(2)
+  const logger = new Logger(apiKey, createOptions())
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger.url)
+    .post('/', () => { return true })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.on('error', (err) => {
+    t.type(err, TypeError, 'Expected to be a TypeError')
+    t.match(err, {
+      name: 'TypeError'
+    , message: 'options parameter must be a level (string), or object'
+    , meta: {
+        got: 'number'
+      }
+    }, 'Expected Error is correct')
+  })
+  logger.log('log line', 12345)
+})
+
+test('.log() passed an invalid log level uses the default instead', (t) => {
+  t.plan(3)
+  const logger = new Logger(apiKey, createOptions({
+    level: 'info'
+  }))
+
+  t.on('end', async () => {
+    nock.cleanAll()
+  })
+
+  nock(logger.url)
+    .post('/', (body) => {
+      const obj = body.ls[0]
+      t.equal(obj.level, 'INFO', 'Default level was used instead')
+      return true
+    })
+    .query(() => { return true })
+    .reply(200, 'Ingester response')
+
+  logger.on('error', (err) => {
+    t.type(err, Error, 'Expected to be an Error')
+    t.match(err, {
```

Comment:
deepEqual

Reply:
Here, I'm testing the error itself, so `deepEqual` is the right choice.  Thanks!